packages/transaction-controller/src/TransactionController.ts

```diff
@@ -1366,12 +1356,13 @@ export class TransactionController extends BaseController<
     const txsToKeep = transactions
       .sort((a, b) => (a.time > b.time ? -1 : 1)) // Descending time order
       .filter((tx) => {
-        const { chainId, networkID, status, txParams, time } = tx;
+        const { chainId, status, txParams, time } = tx;
 
         if (txParams) {
-          const key = `${txParams.nonce}-${
-            chainId ? convertHexToDecimal(chainId) : networkID
-          }-${new Date(time).toDateString()}`;
+          const txChainId = chainId ?? txParams.chainId;
```

Comment:
Both the `chainId` in the metadata and in the params is set using the network state so this fallback shouldn't be needed and may even introduce ambiguity as we sort the transactions before approval.

Reply:
I missed that. so `txMeta.txParams.chainId` will not be available in state